CONTRIBUTING.md

```diff
@@ -1,75 +1,389 @@
-Any words that do not pass spell check will be shown in the logs of
+Any words that do fail spell check are shown in the logs of
 the Github Action. Please either fix the words or add them to the list
-in `Agoric/documentation/.wordlist.txt`. Please maintain the list's alphabetical order when entering new words for the convenience of future maintainers.
+in `Agoric/documentation/.wordlist.txt`. When entering new words, please keep the list 
+in alphabetical order for the convenience of future maintainers.
 
 ![](./contributing-assets/spellcheck-results.png)
 
-## Importing and testing snippets
+### Importing and testing code snippets
+
+**tyg todo: Note: There was a lot of this that I found possibly ambiguous. I've tried
+to specify a few more things to make it clearer, but you should probably review
+this section carefully make sure my interpretations are correct or not**
 
-To import code snippets into the documentation, you can write up the
-code in a separate file under `snippets/`, then include it by adding a
-line like `<<< @/snippets/test-intro-zoe.js#install` to your document.
-The section `test-intro-zoe.js` should be replaced with your
-particular filename. `#install` is an example of a specific code
-region in the file to import. For example, it might look like:
+Code snippets are not short inline code bits like `const x = 2 + 2;`. Or 
+code blocks denoted in Markdown by starting with a line consisting of three backquotes with 
+`js` appended and ending with a line consisting of three backquotes. 
 
+Rather, code snippets are actual development or test code from the Agoric-SDK repo,
+or code held to a similar standard of correctness. They should pass `lint` and run with no
+errors. This ensures our documents use real code that works
+with the current agoric-sdk version (whatever is on master) and is
+not outdated. **tyg todo: I'm not quite seeing this guaranteeing things,
+since we're not extracting the code directly from agoric-sdk. It'd seem
+if we don't check and manually update the snippets file after each upgrade, we risk
+falling out of sync with the agoric-sdk repo.**
+
+To import code snippets into the documentation, create a file
+under `Agoric/documentation/snippets/`. **tyg todo: It appears there should be only one 
+snippets folder for the whole doc repo?** Put your code in the new file.
+Remember, it must be able to run without errors. 
+
+You can make an entire code file, or any part of it, into a snippet (each
+of which can be used multiple times in the docs). Just surround the part
+you want to be a snippet with `#region` and `#endregion` comments:
 ```js
-  // #region install
-  const atomicSwapInstallation = await E(zoe).install(atomicSwapBundle);
-  // #endregion install
+// #region regionName
+...
+// #endregion regionName
 ```
+`regionName` in the above is any name you want to give this snippet.
+
+If you want the whole file to be a snippet, just put the `#region` / `#endregion`
```

Comment:
The whole file will be imported if there's no region mentioned in the import or if the region doesn't exist (dangerous!), so no need to put it at top and bottom - just omit from the import.